### PR TITLE
Various features

### PR DESCRIPTION
**Pull request by dariuschira**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,3 +54,33 @@ By default, stats are logged every 5 seconds, you can adjust that using
...
+Certain platforms such as datadog and influxdb offer tagging options.
+
+To enable tagging set ``STATSD_TAGGING`` to ``True``, it is disabled by default:
+
+    STATSD_TAGGING = False
+
+Then, you can set custom tags using ``STATSD_TAGS``:
+
+    STATSD_TAGS = {
+        'spider_name': True
```
Comment (dariuschira) [inline README.md]: This is wip, I'm trying to figure out a better way of allowing custom tags.
I'm thinking either continue with some defaults (add project name / version / date tags) and/or allow a method/attribute on the spider act on these

Comment (andrewbaxter) [inline README.md]: It's not clear to me from this whether the tag will be `spider_name` (and you're supposed to replace that with the actual spider name) or if `spider_name` is a keyword that automatically inserts the spider name.  If the latter, are there any other keywords?  Also, what does `True` mean here?

File: scrapy_statsd_extension/handlers.py [hunk 2]
```diff
@@ -5,15 +5,56 @@
...
+
+    def not_ignored_field(self, key):
+        for prefix in self.ignored_prefixes:
+            if key.startswith(prefix):
+                return False
+
+        return True
+
+    def has_valid_prefix(self, key):
+        if self.dont_log_all_fields is False:
```
Comment (andrewbaxter) [inline scrapy_statsd_extension/handlers.py]: It might be easier to read if you reverse this bool, and make it `log_all_fields`.

**Review by andrewbaxter — CHANGES_REQUESTED** [review]
The readme is really great, good work there.

File: README.md (hunk 1 above)
Comment (andrewbaxter) [reply README.md]: Ah, didn't see this :sweat_smile:

Comment (andrewbaxter) [reply README.md]: I'm :heavy_plus_sign: for attributes.  I have projects where we may have like a `provider` argument or `base_url` or whatever, which would be helpful to tag on.

**Comment by andrewbaxter** [comment]
One other thought, I think the readme doesn't list what handlers are currently supported.  What about having a section per handler with handler-specific notes, settings, etc?

Also FYI, not sure if it helps, but `.rst` supports a directive that will automatically generate a table of contents which might be useful if the readme gets longer.  I personally can't stand rst though :D.  You can use pandoc to convert md -> rst and just add the contents directive there though (and round trip rst -> md -> rst for future edits :sweat_smile: )

**Comment by dariuschira** [comment]
Thanks for the tip, I din't know much about rst tbh :). I prefered markdown because I find it easier to work with but that's just my opinion so I see no issue with changing it

**Comment by dariuschira** [comment]
> One other thought, I think the readme doesn't list what handlers are currently supported. What about having a section per handler with handler-specific notes, settings, etc?

I think we can do that once we have more than one, so far the docs cover the basic use cases and we can extend later. What do you think?

File: README.md (hunk 1 above)
Comment (dariuschira) [reply README.md]: Added attribute 'statsd_tags' that should be a dictionary with additional tags. I think this is a good starting point and we can extend later.

**Comment by andrewbaxter** [comment]
> I think we can do that once we have more than one, so far the docs cover the basic use cases and we can extend later.

Ah okay, I figured out what was confusing me.  In the docs it says:

```
Certain platforms such as datadog and influxdb offer tagging options.
```
in the tags section, but I was thinking those were separate protocols than statsd.  At least, I thought influxdb had its own protocol.

So would the other providers be for like logstash/prometheus and datadot/influxdb would continue to use the statsd protocol?

**Review by andrewbaxter — APPROVED** [review]
LGTM, thanks for the answers

**Pull request merged**